src/pgduckdb_planner.cpp

```diff
@@ -135,6 +143,35 @@ DuckdbRangeTableEntry(CustomScan *custom_scan) {
 
 PlannedStmt *
 DuckdbPlanNode(Query *parse, int cursor_options, bool throw_error) {
+
+	/* If we're accessing from view, check
+	 * that Postgres user is allowed to access it */
+
+	ListCell *l;
+	foreach (l, parse->rtable) {
```

Comment:
I expect this not to error for views in subqueries? I think you'll need to walk the query tree like is done for `ContainsDuckdbItems`.